contracts/cw-ics721-bridge/src/contract.rs

```diff
@@ -356,9 +362,80 @@ pub fn query(deps: Deps, _env: Env, msg: QueryMsg) -> StdResult<Binary> {
         QueryMsg::Pauser {} => to_binary(&PO.query_pauser(deps.storage)?),
         QueryMsg::Paused {} => to_binary(&PO.query_paused(deps.storage)?),
         QueryMsg::Proxy {} => to_binary(&PROXY.load(deps.storage)?),
+        QueryMsg::Cw721CodeId {} => to_binary(&query_cw721_code_id(deps)?),
+        QueryMsg::ClassIdToNftContract { start_after, limit } => {
+            to_binary(&query_class_id_to_nft_contract(deps, start_after, limit)?)
+        }
+        QueryMsg::OutgoingClassTokenToChannel(query) => to_binary(&query_class_token_to_channel(
+            deps,
+            OUTGOING_CLASS_TOKEN_TO_CHANNEL,
+            query,
+        )?),
+        QueryMsg::IncomingClassTokenToChannel(query) => to_binary(&query_class_token_to_channel(
+            deps,
+            INCOMING_CLASS_TOKEN_TO_CHANNEL,
+            query,
+        )?),
     }
 }
 
+fn query_cw721_code_id(deps: Deps) -> StdResult<u64> {
+    CW721_CODE_ID.load(deps.storage)
+}
+
+fn query_class_id_to_nft_contract(
+    deps: Deps,
+    start_after: Option<ClassId>,
+    limit: Option<u32>,
+) -> StdResult<Vec<ClassIdToNftContractResponse>> {
+    cw_paginate::paginate_map(
+        deps,
+        &CLASS_ID_TO_NFT_CONTRACT,
+        start_after,
+        limit,
+        Order::Ascending,
+    )?
+    .into_iter()
+    .map(|p| {
+        let (class_id, nft_contract) = p;
+        Ok(ClassIdToNftContractResponse {
+            class_id,
+            nft_contract,
+        })
+    })
+    .collect::<StdResult<Vec<ClassIdToNftContractResponse>>>()
+}
+
+fn query_class_token_to_channel(
+    deps: Deps,
+    class_token_to_channel: Map<(ClassId, TokenId), String>,
+    query: ClassTokenToChannelQuery,
+) -> StdResult<Vec<ClassTokenToChannelResponse>> {
+    let start_after = query.start_after.map(|class_token| {
+        (
+            ClassId::new(class_token.class_id),
+            TokenId::new(class_token.token_id),
+        )
+    });
+    cw_paginate::paginate_map(
+        deps,
+        &class_token_to_channel,
+        start_after,
+        query.limit,
+        Order::Ascending,
+    )?
+    .into_iter()
+    .map(|p| {
+        let ((class_id, token_id), channel) = p;
+        Ok(ClassTokenToChannelResponse {
+            class_id,
+            token_id,
+            channel,
+        })
+    })
+    .collect::<StdResult<Vec<ClassTokenToChannelResponse>>>()
```

Comment:
Same comment as above about the map, feels like its not needed to be done on the contract level, but rather be done on the front-end.